nightfall-deployer/README.md

```diff
@@ -0,0 +1,238 @@
+    `amount` is the determined max allowed value for deposits and withdrawals. 
+    In this token list should exist one entry related to the L2 Token that is going to be used to pay fees. After defining it, its ID should be set in the variable `FEE_L2_TOKEN_ID` (e.g. using the sample bellow, if one is going to use `WETH` as the L2 Token for paying fees, then set `FEE_L2_TOKEN_ID=WETH`).
+    The placeholder `NETWORK_NAME` should be replaced with the network name:
+      ```
+        tokens: {
+            [NETWORK_NAME]: [
+            {
+              name: 'MATIC',
+              address: '0x0000000000000000000000000000000000001010',
+              amount: '1000000000000000000000',
+            },
+            ...
+          ],
+          ...
+        }
+      ```
+    Follows a sample:
+      ```
+        tokens: {
+            mumbai: [
+            {
+              name: 'WETH',
+              address: '0xa6fa4fb5f76172d178d61b04b0ecd319c5d1c0aa',
+              amount: '1000000000000000000',
+            },
+            {
+              name: 'MATIC',
+              address: '0x0000000000000000000000000000000000001010',
+              amount: '1000000000000000000000',
+            },
+            {
+              name: 'USDC',
+              address: '0xe6b8a5CF854791412c1f6EFC7CAf629f5Df1c747',
+              amount: '1000000000',
+            },
+            {
+              name: 'stMATIC',
+              address: '0xa337f0B897a874DE1E9F75944629a03F911cFbE8',
+              amount: '1000000000',
+            },
+            ...
+          ],
+          ...
+        }
+      ```
+
+4. Start the deployment with the following command, where the parameter is the .env file created in the `Step 1.`.
+
+    ```
+    ./bin/deploy-contracts [.env.deployment.file]
+    ```
+    e.g.
+    ```
+    ./bin/deploy-contracts .env.deployment.my_network
+    ```
+
+    During the deployment an instance of the `worker` application is required to be up, so that the Circuits can be generated accordingly. This instance is started up 
+    automatically, but it is possible to configure the worker server if one is already available, for this set `CIRCOM_WORKER_HOST` in the .env file.
+
+    There is a variable in the .env file called `DEPLOYMENT_SERVICES`. It allows one to set the services to start up. This variable can be passed over with the command
+    as well.
+    ```
+    DEPLOYMENT_SERVICES=client,optimist,worker ./bin/deploy-contracts .env.deployment.my_network
+    ```
+
+**One should remember to have enough funds when performing the deployment so that it can finish properly.**
+
+After the deployment finishes successfully, one can find the files that were generated during the deployment under the folder `docker/volumes`:
+- `build/contracts`: contain the ABI files.
+- `proving_files`: contain circuits' files.
+
+These files are going to be used by the applications `nightfall-client` & `nightfall-optimist`.
+
```

Comment:
Can you explain how you would import these to `client` and `optimist`, if you were not deploying, but just using the deployment; i.e. you just want to run up a nightfall node?

Reply:
These apps share the folder under `docker/volumes`, where all the proving_files and contracts are persisted. The .env file has an env var called DEPLOYMENT_SERVICES where one can specify the services to start up. I did this as a way to run a quick test if someone desires it.